uk_results/migrations/0030_populate_postresult_post_election.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from collections import defaultdict
+
+from django.db import migrations, models
+
+
+def set_post_election_from_post(apps, schema_editor):
+    """
+    This is far from idea. Try to guess the PostExtraElection
+    that this PostResult relates to. This will have to be done by looking
+    and the related memberships and assuming they're correct (sometimes they
+    won't be, and that will have to be fixed manually later).
+    """
+    PostResult = apps.get_model('uk_results', 'PostResult')
+    PostExtraElection = apps.get_model('candidates', 'PostExtraElection')
+
+    print(PostResult.objects.all().count())
+    qs = PostResult.objects.all().select_related('post__extra')
+    for post_result in qs:
+        pee = None
+        elections = post_result.post.extra.elections.all()
+        if elections.count() == 1:
+            # This is an easy case – this post only has one known election
+            pee = PostExtraElection.objects.get(
+                election=elections.first(),
+                postextra=post_result.post.extra
+            )
+            post_result.post_election = pee
+            post_result.save()
+
+        else:
+            if not post_result.result_sets.exists():
+                # There are no results sets for this post_result
+                # so we cna just delete it
+                post_result.delete()
+                continue
+
+            result_sets_by_election = defaultdict(list)
+            # Work out how many elections we have results for.
+            # If it's only 1, then use that one
+            for result_set in post_result.result_sets.all():
+                for candidate_result in result_set.candidate_results.all():
+                    this_election = candidate_result.membership.extra.election
+                    result_sets_by_election[this_election].append(result_set)
```

Comment:
This code is assuming that the `CandidateResult`s associated with a particular `ResultSet` all have memberships associated with the same `Election`, but unfortunately there are three `ResultSet`s for which that's not the case:

```
for rs in ResultSet.objects.all():
    elections = set()
    for cr in rs.candidate_results.all():
        elections.add(cr.membership.extra.election)
    if len(elections) > 1:
        print "{0} elections associated with {1}".format(len(elections), rs)
   ....:         
2 elections associated with pk=2082 user=None post=<PostResult: PostResult object>
2 elections associated with pk=1569 user=None post=<PostResult: PostResult object>
2 elections associated with pk=1570 user=None post=<PostResult: PostResult object>
```

At least it's only three cases, so I guess it's worth fixing those before running this migration. However, to make sure it really is unambiguous which election a `ResultSet` is for, I'd do a similar check in this bit of code and raise an exception if there were any such cases.